.ci/gitlab/.gitlab-ci.yml

```
@@ -186,10 +209,12 @@ dotenv:
     KUBERNETES_MEMORY_REQUEST: "5M"
   artifacts:
     reports:
-      dotenv: .ci/env
+      dotenv: env
   tags: [ "spack", "service" ]
   script:
-  - cat .ci/env
+  - curl -LSsO https://raw.githubusercontent.com/spack/spack-packages/refs/pull/342/head/.ci/gitlab/scripts/common/write_dotenv.py
+  - python3 write_dotenv.py .ci/env
```

What about something like this to ensure you get the current version from whichever source is appropriate? I really don't like hardcoding the url to the original PR in here.

```suggestion
  - dotenv_py=$(ls $CI_PROJECT_DIR/.ci/gitlab/scripts/common/write_dotenv.py 2>/dev/null || ls $CU_AUX_DIR/.ci/gitlab/scripts/common/write_dotenv.py 2>/dev/null)
  - python3 $dotenv_py .ci/env
```

---

So, like I said, we don't know where this script is in the context of this job. It is possible it is running in the spack/spack context or the spack/spack-packages context and the point is the script determines that and then sets up the subsequent jobs accordingly.

The reason it points to the PR right now is it does not yet exist on develop. Once this is merged we can point it to look at the `HEAD` of develop. I will put in a comment similar to what @alecbcs has for the triage job.

---

That's why my proposal was to check for the script in each context, rather than pulling it from a third source.

---

> Once this is merged we can point it to look at the HEAD of develop.

I'm not really a fan of this either, pulling from a commit where we don't have a guarantee this file wasn't modified unexpectedly for a given PR context (not likely, but it could happen) seems like a recipe for future headaches. 

If we're _never_ using this script from Spack-core/Spack-packages directly, why not put it in a separate repo/gist/whatever rather than have to deal with sync or location issues?

Alternatively is `.ci/gitlab/scripts/common/write_dotenv.py` this the same for both repos? We should know from Gitlab where the clone root is. 

IIRC `.pre` is run after the clone, so that should work right?

---

`CI_AUX_DIR` is determined by that script, it is a chicken-egg problem. This entire step is to setup the context of the pipeline for all of the downstream jobs.

The reason for curl'ing the file directly is to avoid the alternative, which would be to clone the entire develop branch of packages to a known location and then call it from there.

---

So, I am not a huge fan of this either but some of the alternatives I had thought of...
* using a gist
* requiring the script to exist and be mounted for all runners
* doing the same thing, but cloning the entire repo
* encoding the script into a base64 string, then decoding it to a file and execute it with python

We can change this to pull from a commit, but again we can't really change that until _after_ this is merged so that the script lives on a develop commit in `spack/spack-packages`. This is supposed to only be a temporary solution so we can beat the chicken-egg problem.

If it is a huge deal, I can break this script out into another PR, merge that, and then we reference that commit here.

---

As long as we're not pointing at this PR or an endpoint that can change without deliberate intent (or do one of those things now and fix it as soon as this lands), I'm happy, opinions beyond that I will leave to other folks.

(I know I suggested baking into runner images but I was like 80% unserious, that seems like a massive pain to update)